home/import_assessments.py

```diff
@@ -96,9 +98,36 @@ def save_assessments(self) -> None:
             )
 
     def parse_file(self) -> list["AssessmentRow"]:
+        """
+        Converts the iterator to a list of rows.
+        If there are rows:
+            a. Extracts the original headers from the first row.
+            b. Converts the original headers to snake_case and creates a mapping from original headers to snake_case headers.
+            c. Validates that the snake_case headers contain all mandatory headers.
+            d. Transforms each row to use snake_case headers.
+        """
+
+        row_iterator = parse_file(self.file_content, self.file_type)
+        rows = [row for _, row in row_iterator]
+
+        if not rows:
+            raise ImportAssessmentException(
+                "The import file is empty or contains no valid rows.", row_num=1
+            )
+
+        original_headers = rows[0].keys()
+        headers_mapping = {
+            header: self.to_snake_case(header) for header in original_headers
+        }
+        snake_case_headers = list(headers_mapping.values())
+        self.validate_headers(snake_case_headers, MANDATORY_HEADERS, row_num=1)
+        transformed_rows = [
+            {headers_mapping[key]: value for key, value in row.items()} for row in rows
+        ]
```

Comment:
Should some of this not maybe live inside the parse_file function? I see we have a function in there called fix_rows?

I'm not very familiar with all the import code so I'm just asking